src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java

```diff
@@ -42,6 +42,26 @@ private DeathRecord mapDeathRecord(V1Death v1Death) {
         final DeathRecord.DeathRecordBuilder builder = DeathRecord.builder().systemNumber(v1Death.getId());
         final Deceased deceased = v1Death.getDeceased();
 
+        if (nonNull(deceased)) {
+            builder.name(String.format("%s %s", deceased.getForenames(), deceased.getSurname()))
+                .dateOfBirth(deceased.getDateOfBirth())
+                .sex(null == deceased.getSex() ? null : deceased.getSex().getValue())
+                .address(deceased.getAddress())
+                .dateOfDeath(deceased.getDateOfDeath());
+        }
+
+        return builder.build();
+    }
+
+    @SuppressWarnings("squid:S2583")
```

Comment:
is it possible to resolve this warning than suppress

Reply:
This warning is becuase of fields in the api annotated as @NotNull. Becuase the api is outside of out control, and may change I think it is safer to keep the null checks and suppress the warning.